app/models/build.rb

```diff
@@ -0,0 +1,16 @@
+class Build < ActiveRecord::Base
+  SHA1_REGEX = /\A[0-9a-f]{40}/i
+
+  belongs_to :project
+  has_many :statuses, class: 'BuildStatus'
+  has_many :deploys
+  has_many :releases
+
+  validates :project, presence: true
+  validates :git_sha, format: SHA1_REGEX, allow_nil: true
+  validates :container_sha, format: SHA1_REGEX, allow_nil: true
+
+  def successful?
+    statuses.successful.all?
```

Comment:
`statuses.successful` returns either an empty or filled array, so all will always be true

like asking `[<record1>].all?` -> always true